hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/OzoneManager.java

```diff
@@ -3568,7 +3570,7 @@ private DBCheckpoint getDBCheckpointFromLeader(String leaderId) {
         "from the checkpoint.", leaderId);
 
     try {
-      return omSnapshotProvider.getOzoneManagerDBSnapshot(leaderId);
+      return omRatisSnapshotProvider.getOzoneManagerDBSnapshot(leaderId);
```

Comment:
Is it Ok to return null in case of IOException? We don't have a null check where returned value is used at line # 3385-3387. Also the same value is passed to `installCheckpoint` without null check.

Reply:
I didn't write this code, I just renamed the field.  Studying it a bit further, it does look things will work correctly.

installCheckpoint() will throw an exception, which causes installSnapshotFromLeader() to return a null TermIndex which is handled in ratis here:
https://github.com/apache/ratis/blob/release-2.3.0/ratis-server/src/main/java/org/apache/ratis/server/impl/SnapshotInstallationHandler.java#L274-L278

Reply:
Sorry about wrong function name. Inside `installSnapshotFromLeader`, we have following line:

```
DBCheckpoint omDBCheckpoint = getDBCheckpointFromLeader(leaderId);
LOG.info("Downloaded checkpoint from Leader {} to the location {}",
    leaderId, omDBCheckpoint.getCheckpointLocation());
```

It will throw NPE in log statement at `omDBCheckpoint.getCheckpointLocation()`, if `omDBCheckpoint` is null.

Reply:
+1